sites/cheerpx/src/content/docs/11-guides/custom-images.mdx

```diff
@@ -3,36 +3,63 @@ title: Custom disk images
 description: How to create custom ext2 images to use as a filesystem for CheerpX
 ---
 
-This guide will show you how to configure a Docker container and build it into an `ext2` image for use with CheerpX.
+This guide demonstrates how to create a custom `ext2` image for use with CheerpX while ensuring that the root ownership of files is maintained.
 
 ## 1. Create a Dockerfile
 
-For example, the following Dockerfile installs the `curl` command-line tool:
+First, create your `Dockerfile` to define the environment. Here's an example of how to build a Docker image that installs the `curl` command:
 
 ```dockerfile
-FROM --platform=i386 i386/debian:buster
+FROM --platform=i386 docker.io/i386/debian:buster
 ARG DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && apt-get -y upgrade && apt-get -y install curl
 ```
 
-Note that the base image's architecture must be 32-bit x86 (for example, `i386` or `i686`).
+> [!note] Note
+> The base image's architecture must be 32-bit x86 (for example, `i386` or `i686`).
 
 ## 2. Create a container out of your Dockerfile
 
+Once the `Dockerfile` is ready, build the image. Use `buildah` to create a container image for your filesystem.
+
 ```bash
 buildah build -f Dockerfile --platform linux/i386 -t cheerpximage
+```
+
+## 3. Create a container from the Docker Image
+
+Now, create a container from the image using `podman`. Ensure the image name matches the on you built in the previous step.
+
+```bash
 podman create --name cheerpxcontainer cheerpximage
 ```
 
-## 3. Copy the filesystem from the container into a local directory
+## 4. Copy the Filesystem from the Container into a Local directory
+
+Create the local directory for the filesystem:
 
 ```bash
 mkdir cheerpXFS
+```
+
+Now copy the files from the container into the directory. This should preserve root ownership when executed from within a root user environment.
+
+```bash
 podman cp cheerpxcontainer:/ cheerpXFS/
 ```
 
-## 4. Create an ext2 image from the specified directory
+If you are running `podman cp` as a non-root user and face any issues with permissions, you might need to adjust the directory ownership after the copy by using `chown`:
+
+```bash
+sudo chown -R root:root cheerpXFS
```

Comment:
No, this is not the right solution. There might be files owned by different users for valid reasons, including system users or the `user` user we add to our dockerfiles.

As discussed with Yuri in a sync-in the command must be run with root privileges, either by using `sudo` / `su` or by using `podman unshare`. See the build script Yuri added to the Asana task.